communication/client/client.py

```diff
@@ -1,55 +1,85 @@
 import socket
-class VirtualVehicle:
-
+import time
 
-    def __init__(self, server_ip = "0.0.0.0", server_port = 65432):
 
+class VirtualVehicle:
+    def __init__(
+        self,
+        server_ip="0.0.0.0",
+        server_port=65432,
+        agx_ip="0.0.0.0",
+        agx_port=59999,
+    ):
         self.SERVER_IP = server_ip
-        self.PORT = server_port
-        self.ADDR = (self.SERVER_IP, self.PORT)
-        self.MSG_LENGTH = 1024
-
-        self.s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        
+        self.SERVER_PORT = server_port
+        self.SERVER_ADDR = (self.SERVER_IP, self.SERVER_PORT)
         self.message_to_server = ["HELLO", "RESULTS", "GOODBYE"]
+        self.message_from_server = ["DO_TASK", "SESSION DONE"]
 
-    def run_client(self):
-        self.s.connect(self.ADDR)
+        self.AGX_IP = agx_ip
+        self.AGX_PORT = agx_port
+        self.AGX_ADDR = (self.AGX_IP, self.AGX_PORT)
+        self.message_to_agx = ["HELLO", "TRAIN"]
+        self.message_from_agx = ["TASK_SCHEDULED", "RESULTS"]
 
-        self.s.send(self.message_to_server[0].encode("utf-8"))
+        self.MSG_LENGTH = 1024
+        self.socket_server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.socket_agx = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+    def run_client(self):
+        self.socket_server.connect(self.SERVER_ADDR)
+        self.socket_server.send(self.message_to_server[0].encode("utf-8"))
 
         while True:
 
-            data = self.s.recv(self.MSG_LENGTH)
+            data = self.socket_server.recv(self.MSG_LENGTH)
 
-            if str(data, "utf-8") == "TRAIN":
+            if str(data, "utf-8") == self.message_from_server[0]:
                 self._client_do_task(str(data, "utf-8"))
 
-            elif str(data, "utf-8") == "SESSION DONE":
-                print("[CLIENT] Received msg from server: SESSION DONE")
-                print("[CLIENT] Terminating session...")
+            elif str(data, "utf-8") == self.message_from_server[1]:
+                print(
+                    f"[CLIENT] Received msg from server: {self.message_from_server[1]}"
+                )
+                print("[CLIENT] Terminating server session...")
                 break
 
-        self.s.close()
-        print("[CLIENT] Socket closed.")
+        self.socket_server.close()
+        print("[CLIENT] Server socket closed.\n")
 
     def _client_do_task(self, server_message):
```

Comment:
```suggestion
    def _train(self, server_message):
```
maybe we could rename this method now to something more specific?